### ISSUE TITLE
year range wrong when one of the songs has a null value for year

### ISSUE THREAD

**Issue by certuna** [deluan/navidrome#173]
Case: album with songs from multiple years
- if all tracks have year tags, navidrome correctly shows the years range ("1991-2004")
- if one of the songs has year = empty, navidrome shows only the highest year ("2004")
Can you reproduce this?

I also have a couple of albums (regular ones, all tracks same year) where the year is tagged, but navidrome (ffmpeg?) doesn't pick up the year. I'll see if i can dig a bit deeper if there's a specific combo of id3 tag version or a case of v1+v2 that triggers this.

**Comment by deluan** [comment]
I think I can reproduce the first issue here. Will try later tonight. 

For the second issue, can you post the output of `ffmpeg -i yourfile.mp3 -f ffmetadata` ?

**Comment by certuna** [comment]
I've found the problem with the second issue:
```
Input #0, mp3, from '/Volumes/Drive/Music/Artist - 2019 - Album/02 Track.mp3':
  Metadata:
    TYER            : 2019-03-18
```
TYER is an id3v2.3 field that should only have 4 digits so whatever software wrote the tags screwed up.

I've got another case, where navidrome shows the artist as [Unknown Artist] even though it does have an Artist tag. Investigating that with `ffmpeg -f ffmetadata` shows that ffmpeg returns no artist field at all, so this is either an ffmpeg bug or another malformed tag. I guess I just rebuild the tags and try again.

**Comment by deluan** [comment]
For the year issue, you can send me an example file and I can add support for it (adding `TYER` to the list of recognizable `year` fields. 

For the artist issue, if `ffmpeg` cannot find any artist (or equivalent) tag, unfortunately there's nothing we can do, maybe open an issue with `ffmpeg` project.

**Comment by certuna** [comment]
No this is a 'problem' with ffmpeg. By the looks of it:
- if ffmpeg finds a correct 4-digit id3v2.3 TYER tag, it outputs it with the field name `year` (which navidrome picks up correctly)
- if ffmpeg finds off-spec values in that TYER tag (ie, a full yyyy-mm-dd date), it outputs it with the raw tag name `TYER`
I don't think navidrome needs to do anything.

Agree that the artist issue has nothing to do with navidrome. I've rebuilt the tags and everything works.

**Comment by deluan** [comment]
But this case of `TYER` having an invalid format, and not being parsed into `year` could be handled by Navidrome, similar to what I do with the composer tag:

https://github.com/deluan/navidrome/blob/master/scanner/metadata_ffmpeg.go#L31

Don't you think it is a good idea to handle this edge case?

**Comment by deluan** [comment]
Regarding the original issue:
> if one of the songs has year = empty, navidrome shows only the highest year ("2004")

I can reproduce this, but my question is: Do we want to fix this, by ignoring the empty year? Isn't it an issue with the album tagging that should be fixed?

**Comment by deluan** [comment]
Fixed in 69dc4d97b3a6e510509b6db1305a43bea86fa309. I'll close this, but feel free to chime in if you still see any related problems.

**Comment by certuna** [comment]
> Regarding the original issue:
> 
> > if one of the songs has year = empty, navidrome shows only the highest year ("2004")
> 
> I can reproduce this, but my question is: Do we want to fix this, by ignoring the empty year? Isn't it an issue with the album tagging that should be fixed?

Sorry, forgot to reply here :) I would say yes, ignore the empty year and show a range for the rest of the tracks. For an album with, say 20 tracks with years and one with the year missing, that screws up a (quite informative) data range for the rest.
